llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1750,6 +1752,89 @@ void lowerGlobalsToVector(Module &M) {
 
 } // namespace
 
+static void checkSLMInit(Module &M) {
+  SmallPtrSet<const Function *, 8u> Callers;
+  bool KernelHasSLMInit = false;
+  bool KernelHasLocalAccessor = false;
```

Comment:
The function accepts a module `M`. The vars are initialized to `false` once here. If so, the vars should be named not `KernelHasSLMInit`, but `ModuleHasSLMInit`. Same for the 2nd var.